### PR TITLE
feat: Windows Support

### PR DESCRIPTION
**Pull request by alvindaiyan**
## Description

This pr is for supporting windows usage of the extension. Mostly, removed linux command line usage and paths, replaced with python native libraries. 

Fixes # (issue)

Windows Compatibility support

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: build_scripts/training/sagemaker_entrypoint.py [hunk 1]
```diff
@@ -98,19 +100,16 @@ def upload_model_to_s3_v2(model_name, s3_output_path, model_type):
...
                     output_tar = file
                     tar_command = f"tar cvf {output_tar} {safetensors} {yaml}"
-                    print(tar_command)
-                    os.system(tar_command)
-                    logger.info(f"Upload check point to s3 {output_tar} {output_bucket_name} {s3_output_path}")
-                    print(f"Upload check point to s3 {output_tar} {output_bucket_name} {s3_output_path}")
-                    upload_file_to_s3(output_tar, output_bucket_name, os.path.join(s3_output_path, model_name))
                 elif model_type == "Lora":
                     output_tar = file
                     tar_command = f"tar cvf {output_tar} {safetensors}"
```
Comment (guming3d) [inline build_scripts/training/sagemaker_entrypoint.py:105]: tar_command is no longer used, pls delete it

File: utils.py [hunk 2]
```diff
@@ -235,6 +244,215 @@ def get_variable_from_json(variable_name, filename='sagemaker_ui.json'):
...
+    if src_path.exists():
+        if dest_path.exists() and not force:
+            raise FileExistsError(f"Destination path '{dest}' already exists and 'force' is not set")
+        else:
+            if dest_path.is_file():
+                dest_path.unlink()
+            elif dest_path.is_dir():
+                shutil.rmtree(dest_path)
+
+        if src_path.is_file() or src_path.is_dir():
```
Comment (guming3d) [inline utils.py:408]: just curious that why we need to add checking here? Is it possible that src_path can be neither file or dir?

File: build_scripts/training/sagemaker_entrypoint.py (hunk 1 above)
Comment (alvindaiyan) [reply build_scripts/training/sagemaker_entrypoint.py:105]: just for debugging purpose, once stable, can delete

File: utils.py [hunk 3]
```diff
@@ -235,6 +244,215 @@ def get_variable_from_json(variable_name, filename='sagemaker_ui.json'):
...
+        filesystems = df(show_all=True, human_readable=True)
+        for filesystem in filesystems:
+            print(f"Filesystem: {filesystem['filesystem']}")
+            print(f"Total: {filesystem['total']}")
+            print(f"Used: {filesystem['used']}")
+            print(f"Free: {filesystem['free']}")
+            print(f"Percent: {filesystem['percent']}%")
+            print(f"Mountpoint: {filesystem['mountpoint']}")
+    """
+    partitions = psutil.disk_partitions(all=show_all)
```
Comment (guming3d) [inline utils.py:439]: Make sure the psutil can get filesystem correctly in windows os.

**Review by guming3d — APPROVED** [review]
Just have some minor comments, please go ahead and ship it.

**Review by NingLu — APPROVED** [review]

**Pull request merged**